openmc/universe.py

```diff
@@ -76,10 +76,17 @@ def __eq__(self, other):
             return False
         elif self.name != other.name:
             return False
-        elif self.cells != other.cells:
+        elif len(self.cells) != len(other.cells):
             return False
-        else:
-            return True
+
+        # Check each cell
+        for cell_id in self.cells:
+            if cell_id not in other.cells:
+                return False
+            if self.cells[cell_id] != other.cells[cell_id]:
+                return False
```

Comment:
I was curious about this so I dug a little bit. [Here](https://github.com/python/cpython/blob/master/Objects/dictobject.c#L2692-L2737) is the Python `dict.__eq__` implementation. It checks the length of each, then compares keys/values (much the same as you are doing here). However, `Universe.cells` is an `OrderedDict` -- the check for equality on those also [accounts for the order](https://github.com/python/cpython/blob/master/Lib/collections/__init__.py#L291). So, my guess is that when you were converting back and forth, the order of cells ended up different. If you want to do an order-insensitive equality check here, you could just do:
```Python
dict.__eq__(self.cells, other.cells)
```
Can you check if that works for the TREAT model conversion?